src/utils/jwt-utils.js

```diff
@@ -20,12 +20,29 @@ const jwtUtil = {
       expiresIn: AUTH_TOKEN_EXPIRY_MS,
     })
   ),
-  encryptToken: _.wrap(AES.encrypt, (encrypt, token) =>
-    encrypt(token, ENCRYPTION_SECRET).toString()
-  ),
-  decryptToken: _.wrap(AES.decrypt, (decrypt, token) =>
-    decrypt(token, ENCRYPTION_SECRET).toString(CryptoJS.enc.Utf8)
-  ),
+  encryptToken: (token) => {
+    // NOTE: The iv here is 16 characters long.
+    // This is because we generate 8 random bytes (1 byte = 8 bits)
+    // but hex is 4 bits per character..
+    const iv = crypto.randomBytes(8).toString("hex")
```

Comment:
hmm i think this wont work, we need the same iv over all the instances!
what we want here is an env var rather than creating this randomly at runtime!

Reply:
we don't want an env var - the iv needs to be unpredictable and unique (best is cryptographically random, see [here](https://en.wikipedia.org/wiki/Initialization_vector))

this works by appending the iv itself to the encrypted token and using it at the decrypt stage to check.

<img width="974" alt="Screenshot 2024-03-12 at 6 50 17 PM" src="https://github.com/isomerpages/isomercms-backend/assets/44049504/58a31410-4eb3-4144-98a0-00c4f9660805">

Reply:
hmm but there is a chance that the encrypt and decrypt steps occur in different different instances? with the different values over say two instances, how will this work ah?

Reply:
Initialisation vectors **must be randomized** otherwise it breaks the security of encryption. The initialisation vector is **prepended** to the cipher text, which the decryption then uses to initialise itself as well.
https://knowledge-base.secureflag.com/vulnerabilities/broken_cryptography/reused_iv_key_pair_vulnerability.html:

> The consequences of reusing the same IV-Key pair depend on the actual cipher and mode used. In the case of Cipher Block Chaining (CBC) and Cipher Feedback (CFB) modes, the leaks of information from the first block and commonalities are enough to lead an empowered attacker to fully compromising the encryption layer of protection in the given system.

Reply:
To emphasise: the initialisation vector is randomised but not encrypted, it's just prepended before the actual encrypted cipher text. It's used to initialise the cipher randomly otherwise it's vulnerable to typical bit flipping and known planet attacks. So there's no problem across different instances, cos the decrypting cipher only needs to take it from the prepended IV in the cipher text.